qiskit/transpiler/__init__.py

```diff
@@ -40,629 +41,787 @@
 
-   The choice of `initial_layout` can mean the difference between getting a result,
-   and getting nothing but noise.
+.. image:: /source_images/mapping.png
 
-   Let's see what layouts are automatically picked at various optimization levels.  The modified
-   circuits returned by :func:`qiskit.compiler.transpile` have this initial layout information
-   in them, and we can view this layout selection graphically using
-   :func:`qiskit.visualization.plot_circuit_layout`:
+By default, qiskit will do this mapping for you.  The choice of mapping depends on the
+properties of the circuit, the particular device you are targeting, and the optimization
+level that is chosen The choice of initial layout is extremely important when:
 
-   .. jupyter-execute::
+1. Computing the number of SWAP operations needed to map the input circuit onto the device
+   topology.
 
-      from qiskit import QuantumCircuit, transpile
-      from qiskit.visualization import plot_circuit_layout
-      from qiskit.providers.fake_provider import FakeVigo
-      backend = FakeVigo()
+2. Taking into account the noise properties of the device.
 
-      ghz = QuantumCircuit(3, 3)
-      ghz.h(0)
-      ghz.cx(0,range(1,3))
-      ghz.barrier()
-      ghz.measure(range(3), range(3))
-      ghz.draw(output='mpl')
+The choice of `initial_layout` can mean the difference between getting a result,
+and getting nothing but noise. Due to the importance of this stage the preset pass managers
+try a few different methods to find the best layout. Typically this involves 2 stages, first
+trying to find a "perfect" layout (a layout which does not require any SWAP operations) and
+a heuristic pass that tries to find the best layout to use if a perfect layout can not be found.
+For the first stage there are 2 passes typically used for this:
 
+- :class:`~.VF2Layout`: Models layout selection as a subgraph isomorphism problem and tries
+  to find a subgraph of the connectivity graph that is isomporphic to the
+  graph of 2 qubit interactions in the circuit. If more than isomorphic mapping is found a
+  scoring heuristic is run to select the mapping which would result in the lowest average error
+  when executing the circuit.
 
-   - **Layout Using Optimization Level 0**
+- :class:`~.TrivialLayout`: Map virtual qubits to the same numbered physical qubit on the device,
+  i.e. ``[0,1,2,3,4]`` -> ``[0,1,2,3,4]``. This is only used in ``optimization_level=1`` to find a
+  perfect layout and if it fails :class:`~.VF2Layout` is used instead).
 
-      .. jupyter-execute::
+Then for the heuristic pass 2 passes are used by default:
 
-         new_circ_lv0 = transpile(ghz, backend=backend, optimization_level=0)
-         plot_circuit_layout(new_circ_lv0, backend)
+- :class:`~.SabreLayout`: Used to select a layout if a perfect layout isn't found for optimization
+  levels 1, 2, and 3.
+- :class:`~.TrivialLayout`: Always used for the layout at optimization level 0
```

Comment:
Technically there's three, because O1 with control flow uses `DenseLayout`.

Reply:
Ah good point, I'll add `DenseLayout` here too so we can backport this change for 0.22.3 (assuming we'll push one). When we make sabre control flow aware we'll have to remember to update this.

Reply:
Added in: https://github.com/Qiskit/qiskit-terra/pull/9087/commits/a46e97de3261a7a394c0f63ff23f818f5936f49f